### PR TITLE
Adjust to recent printing changes

### PR DESCRIPTION
**Pull request by thofma**
This makes the recent "pretty printing" changes available in Oscar, which were implemented here: https://github.com/Nemocas/AbstractAlgebra.jl/pull/1320.

I did not change the printing of any Oscar types. The diff comes only from the changes upstream.

CC: @simonbrandhorst 

Edit: I copied the docs over so that one sees the `Indent()` etc in action.

### CODE REVIEW

**Comment by thofma** [comment]
Hm, I found a bug in the printing. I will have to fix this first.

**Comment by thofma** [comment]
Should be good to

File: docs/src/CommutativeAlgebra/rings.md [hunk 1]
```diff
@@ -119,67 +119,71 @@ Gröbner and standard bases are implemented for multivariate polynomial rings ov
 
 ```jldoctest
 julia> QQ
-Rational Field
+Fraction field
+  of integer Ring
```
Comment (simonbrandhorst) [inline docs/src/CommutativeAlgebra/rings.md]: Is this intended?

Comment (thofma) [reply docs/src/CommutativeAlgebra/rings.md]: Good catch! Fixed

**Review by simonbrandhorst — APPROVED** [review]

**Comment by thofma** [comment]
Any objections to merging this quickly @fingolfin @fieker?

**Comment by fieker** [comment]
On Fri, May 12, 2023 at 02:03:39PM -0700, Tommy Hofmann wrote:
> Any objections to merging this quickly @fingolfin @fieker?
Be my guest and do it

**Pull request merged**